src/pages/Lending/Pool/components/Borrow/BorrowSweep.tsx

```diff
@@ -0,0 +1,251 @@
+  }, [history])
+  const divider = useMemo(() => <Divider />, [])
+
+  const collateralAccountFilter = useMemo(
+    () => ({ accountId: collateralAccountId }),
+    [collateralAccountId],
+  )
+  const collateralAccountMetadata = useAppSelector(state =>
+    selectPortfolioAccountMetadataByAccountId(state, collateralAccountFilter),
+  )
+
+  const getBorrowFromAddress = useCallback(() => {
+    if (!(wallet && collateralAccountMetadata)) return null
+    return getThorchainFromAddress({
+      accountId: collateralAccountId,
+      assetId: collateralAssetId,
+      getPosition: getThorchainLendingPosition,
+      accountMetadata: collateralAccountMetadata,
+      wallet,
+    })
+  }, [wallet, collateralAccountId, collateralAssetId, collateralAccountMetadata])
+
+  useEffect(() => {
+    if (fromAddress) return
+    ;(async () => {
+      const _fromAddress = await getBorrowFromAddress()
+      if (!_fromAddress) return
+      setFromAddress(_fromAddress)
+    })()
+  }, [getBorrowFromAddress, fromAddress])
+
+  const { data: estimatedFeesData, isLoading: isEstimatedFeesDataLoading } =
+    useGetEstimatedFeesQuery({
+      cryptoAmount: '0',
+      assetId: collateralAssetId,
+      to: fromAddress ?? '',
+      sendMax: true,
+      accountId: collateralAccountId,
+      contractAddress: undefined,
+      enabled: true,
+    })
+
+  const handleSweep = useCallback(async () => {
+    if (!wallet) return
+
+    setIsSweepPending(true)
+
+    try {
+      const fromAddress = await getBorrowFromAddress()
+      if (!fromAddress)
+        throw new Error(`Cannot get from address for accountId: ${collateralAccountId}`)
+      if (!estimatedFeesData) throw new Error('Cannot get estimated fees')
+      const sendInput = {
+        accountId: collateralAccountId,
+        to: fromAddress,
+        input: fromAddress,
+        assetId: collateralAssetId,
+        from: '',
+        cryptoAmount: '0',
+        sendMax: true,
+        estimatedFees: estimatedFeesData.estimatedFees,
+        amountFieldError: '',
+        fiatAmount: '',
+        vanityAddress: '',
+        feeType: FeeDataKey.Fast,
+        fiatSymbol: '',
+      }
+
+      const txId = await handleSend({ wallet, sendInput })
+      setTxId(txId)
+    } catch (e) {
+      console.error(e)
+    }
+  }, [collateralAccountId, collateralAssetId, estimatedFeesData, getBorrowFromAddress, wallet])
+
+  const feeAsset = useAppSelector(state =>
+    selectFeeAssetByChainId(state, fromAssetId(collateralAssetId).chainId),
+  )
+
+  const adapter = getChainAdapterManager().get(fromAssetId(feeAsset?.assetId ?? '').chainId)
+
+  const { poll } = usePoll()
+  useEffect(() => {
+    if (!adapter || !fromAddress) return
+    // Once we have a Txid, the Tx is in the mempool which is enough to broadcast the actual Tx
+    // but we still need to double check that the matching UTXO is seen to ensure coinselect gets fed the right UTXO data
+    if (!txId) return
+    ;(async () => {
+      await poll({
```

Comment:
this poll does provide the detection logic required, but without passing these utxos to the confirm for building the send tx, it still leaves open the chance for the subsequent call to `getUtxos` in `buildSendApiTransaction` to get the utxo list not containing the new unconfirmed utxo. (ex. Node A broadcast, Poll Node A and see unconfirmed utxo, build send tx and query Node B for utxos and they haven't propagated yet)

Reply:
As per dm with @gomesalexandre, going to leave this for a follow up